File: 1833-maximum-ice-cream-bars/1833-maximum-ice-cream-bars.js
/**
 * @param {number[]} costs
 * @param {number} coins
 * @return {number}
 */
var maxIceCream = function(costs, coins) {
    let result = 0;
    costs.sort((a, b) => a - b);
    
    for (const cost of costs) {
        if (cost > coins) {
            break;
        }
        result += 1;
        coins -= cost;
    }
    return result;
};

File: 1833-maximum-ice-cream-bars/1833-maximum-ice-cream-bars.py
class Solution:
    def maxIceCream(self, costs: List[int], coins: int) -> int:
        sorted_costs = sorted(costs)
        cur_coins = coins
        result = 0
        for cost in sorted_costs:
            if cur_coins < cost:
                break
            cur_coins -= cost
            result += 1

        return result